agent/agent.go

```diff
@@ -557,53 +626,19 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext) (shouldExit bool,
 		err = a.logPanic(tc.logger, pErr, err, op)
 	}()
 
-	// If the heartbeat aborts the task immediately, we should report that
-	// the task failed during initial task setup.
-	factory, ok := command.GetCommandFactory("setup.initial")
-	if !ok {
-		return false, errors.New("setup.initial command is not registered")
-	}
-	tc.setCurrentCommand(factory())
-
-	var taskConfig *internal.TaskConfig
-	taskConfig, err = a.makeTaskConfig(ctx, tc)
-	if err != nil {
-		err = errors.Wrap(err, "making task config")
-		grip.Error(err)
-		grip.Infof("Task complete: '%s'.", tc.task.ID)
-		tc.logger = client.NewSingleChannelLogHarness("agent.error", a.defaultLogger)
-		return a.handleTaskResponse(tskCtx, tc, evergreen.TaskSystemFailed, err.Error())
+	// setup should happen before the heartbeat so that it's not abortable
```

Comment:
`// Setup occurs before the task is actually running, so it's not abortable. If setup is taking a long time, timing it out after the heartbeat timeout ensures the operation will give up within the same timeout as the usual amount of time a task is allowed to run before it must heartbeat.`